src/components/SubRequirement.vue

```diff
@@ -88,10 +100,60 @@ import dropupCompletedSrc from '@/assets/images/dropup-lightgray.svg';
 import dropdownIncompleteSrc from '@/assets/images/dropdown.svg';
 import dropdownCompletedSrc from '@/assets/images/dropdown-lightgray.svg';
 
+import {
+  FirestoreSemesterCourse,
+  AppCourse,
+  firestoreCourseToAppCourse
+} from '@/user-data';
+
 Vue.component('completedsubreqcourse', CompletedSubReqCourse);
 Vue.component('incompletesubreqcourse', IncompleteSubReqCourse);
 
+require('firebase/functions');
+
+const functions = firebase.functions();
+
+const FetchCourses = firebase.functions().httpsCallable('FetchCourses');
+
+type CrseInfo = {
+  roster: string;
+  crseIds: number[];
+}
+
+type Data = {
+  showFulfillmentOptionsDropdown: boolean;
+  subReqCoursesNotTakenArray: CrseInfo[][];
+  subReqCourseObjectsNotTakenArray: AppCourse[];
+  dataReady: boolean;
+}
+
 export default Vue.extend({
+  mounted() {
+    if (this.subReq.requirement.courses) {
```

Comment:
Can you avoid this kind of existence check? Can we make sure ensure it always exists?